hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/tasks/TestNSSummaryUnifiedControl.java

```diff
@@ -44,100 +51,176 @@
+
+    ReconContainerMetadataManager reconContainerMgr = mock(ReconContainerMetadataManager.class);
+    ReconGlobalStatsManager reconGlobalStatsManager = mock(ReconGlobalStatsManager.class);
+    ReconFileMetadataManager reconFileMetadataManager = mock(ReconFileMetadataManager.class);
+
+    taskController = new ReconTaskControllerImpl(ozoneConfiguration, new HashSet<>(),
+        mockTaskStatusUpdaterManager, reconDbProvider, reconContainerMgr, mockNamespaceSummaryManager,
+        reconGlobalStatsManager, reconFileMetadataManager);
+
+    taskController.registerTask(nsSummaryTask);
+
+    // Setup mock OM metadata manager with checkpoint support
+    setupMockOMMetadataManager();
+    taskController.updateOMMetadataManager(mockReconOMMetadataManager);
+
+    // Setup successful rebuild by default
+    doNothing().when(mockNamespaceSummaryManager).clearNSSummaryTable();
+
+    // Start async processing
+    taskController.start();
   }
 
   @AfterEach
   void tearDown() {
-    // Reset static state after each test to ensure test isolation
+    // Reset static state after each test
     NSSummaryTask.resetRebuildState();
+
+    // Shutdown task controller
+    if (taskController != null) {
+      taskController.stop();
+    }
   }
-  
-  /**
-   * Create a testable NSSummaryTask that uses mocked sub-tasks for successful execution.
-   */
+
+  private void setupMockOMMetadataManager() throws IOException {
+    DBStore mockDBStore = mock(DBStore.class);
+    File mockDbLocation = mock(File.class);
+    DBCheckpoint mockCheckpoint = mock(DBCheckpoint.class);
+    Path mockCheckpointPath = Paths.get("/tmp/test/checkpoint");
+
+    when(mockReconOMMetadataManager.getStore()).thenReturn(mockDBStore);
+    when(mockDBStore.getDbLocation()).thenReturn(mockDbLocation);
+    when(mockDbLocation.getParent()).thenReturn("/tmp/test");
+    when(mockDBStore.getCheckpoint(anyString(), any(Boolean.class))).thenReturn(mockCheckpoint);
+    when(mockCheckpoint.getCheckpointLocation()).thenReturn(mockCheckpointPath);
+
+    ReconOMMetadataManager mockCheckpointedManager = mock(ReconOMMetadataManager.class);
+    when(mockCheckpointedManager.getStore()).thenReturn(mockDBStore);
+    when(mockReconOMMetadataManager.createCheckpointReconMetadataManager(any(), any()))
+        .thenReturn(mockCheckpointedManager);
+  }
+
   private NSSummaryTask createTestableNSSummaryTask() {
     return new NSSummaryTask(
-        mockNamespaceSummaryManager, 
-        mockReconOMMetadataManager, 
+        mockNamespaceSummaryManager,
+        mockReconOMMetadataManager,
         ozoneConfiguration) {
-      
+
       @Override
       public TaskResult buildTaskResult(boolean success) {
         return super.buildTaskResult(success);
       }
-      
-      @Override 
+
+      @Override
+      public NSSummaryTask getStagedTask(ReconOMMetadataManager stagedOmMetadataManager,
+                                         DBStore stagedReconDbStore) throws IOException {
```

Comment:
nit: Please do not format method signature like this. Whenever visibility / return type / method name / other modifiers are changed, we would have to reindent all parameters.